RIOT-OS/RIOT
# drivers: support for NXP PCA9685 I2C 16-channel, 12-bit PWM controller

### Contribution description

This PR adds a driver vor the NXP PCA9685 16-channel, 12-bit PWM controller that is connected via I2C. Although the controller is optimized for LED control, the 12-bit resolution also allows the control of servos with a resolution of 4 us at 60 Hz refresh signal.

The following features of the PCA9685 are supported by the driver:

- 16 channels with 12-bit resolution
- Refresh rates from 24 Hz to 1526 Hz with internal 25 MHz oscillator
- Totem pole outputs with 25 mA as sink and 10 mA as source at 5V
- Software programmable open-drain output selection
- Inverted outputs
- Active LOW Output Enable (OE) input pin
- External clock input with max. 50 MHz

The driver interface is kept as compatible as possible with the peripheral PWM interface. The only differences are that

- functions have the prefix `pca9685_` and
- functions require an additional parameter, the pointer to the PWM device of type #pca9685_t.

**Update:**
~The driver is also compatible with the PWM extension API in PR #10533 so that up to 64 PCA9685 PWM devices can be used as PWM extensions with the standard peripheral PWM interface.~
To get it merged soon, all PWM extension API code was removed.

### Testing procedure

Please refer the test application in `tests/driver_pca9685` for an example on how to use the driver 
```
make -C tests/driver_pca9685 BOARD=... flash term
```
Connect a logic analyzer and execute the following tests:

1. The signals should alter with frequency of 100 Hz and the duty cycles should be left aligned.
   ```
   init 0 0 100 1000
   set 0 0 100
   set 0 1 900
   ```
2. The signals should alter with frequency of 500 Hz and the duty cycles should be right aligned.
   ```
   init 0 1 500 100
   set 0 0 10
   set 0 1 90
   ```
3. The signals should alter with frequency of 1000 Hz and the duty cycles should be center aligned.
   ```
   init 0 2 1000 100
   set 0 0 10
   set 0 1 90
   ```

### Issues/PRs references

~The PR is related to #10533 but does not depend on it. As long as module `extend_pwm` is not enabled, it is working without #10533.~
